src/kernel/kmain.zig

```diff
@@ -62,27 +61,26 @@ var kernel_heap: heap.FreeListAllocator = undefined;
 
 export fn kmain(boot_payload: arch.BootPayload) void {
     const serial_stream = serial.init(boot_payload);
-
     log_root.init(serial_stream);
 
     const mem_profile = arch.initMem(boot_payload) catch |e| {
         panic_root.panic(@errorReturnTrace(), "Failed to initialise memory profile: {}", .{e});
     };
     var fixed_allocator = mem_profile.fixed_allocator;
 
-    panic_root.init(&mem_profile, &fixed_allocator.allocator) catch |e| {
-        panic_root.panic(@errorReturnTrace(), "Failed to initialise panic: {}\n", .{e});
-    };
-
-    pmm.init(&mem_profile, &fixed_allocator.allocator);
-    var kernel_vmm = vmm.init(&mem_profile, &fixed_allocator.allocator) catch |e| {
+    pmm.init(&mem_profile, fixed_allocator.allocator());
+    var kernel_vmm = vmm.init(&mem_profile, fixed_allocator.allocator()) catch |e| {
         panic_root.panic(@errorReturnTrace(), "Failed to initialise kernel VMM: {}", .{e});
     };
 
-    kmain_log.info("Init arch " ++ @tagName(builtin.arch) ++ "\n", .{});
+    kmain_log.info("Init arch " ++ @tagName(builtin.cpu.arch) ++ "\n", .{});
     arch.init(&mem_profile);
     kmain_log.info("Arch init done\n", .{});
 
+    panic_root.init(&mem_profile, fixed_allocator.allocator()) catch |e| {
+        panic_root.panic(@errorReturnTrace(), "Failed to initialise panic: {}\n", .{e});
+    };
+
```

Comment:
Given this is now move lower down, there are still calls to panic above. If these `panic` calls are called, what will happen? Will it just print address and no symbols. Might be worth changing the `init` function to `init_symbols` or something similar to indicate that this doesn't initialise the panic handler, but initialises the symbol mappings.

Reply:
Yeah if something above this panics then it will just print the addresses without symbols. I like that idea, it makes the purpose clear.